src/Raven.Client/Http/ServerNode.cs

```diff
@@ -36,10 +39,35 @@ public override int GetHashCode()
         {
             unchecked
             {
-                var hashCode = Url?.GetHashCode() ?? 0;
-                hashCode = (hashCode * 397) ^ (Database?.GetHashCode() ?? 0);
-                return hashCode;
+                return HashCombiner.CombineInline(Url?.GetHashCode() ?? _emptyStringHash, Database?.GetHashCode() ?? _emptyStringHash);
             }
         }
+
+        private int _lastServerVersionCheck = 0;
```

Comment:
Can kill this, I think. The failover discard should be enough

Reply:
Yes, but we negotiated a tradeoff with @ppekrol about how to deal with that... for now we will keep it and if eventually, we find we don't need the 'rechecks' then we will remove that part of the checks.